addons/misra.py

```diff
@@ -1107,11 +1150,17 @@ def misra_13_6(data):
 
 def misra_14_1(data):
     for token in data.tokenlist:
-        if token.str != 'for':
-            continue
-        exprs = getForLoopExpressions(token)
-        if exprs and hasFloatComparison(exprs[1]):
-            reportError(token, 14, 1)
+        if token.str != 'for' and token.str != 'while':
```

Comment:
You do not need to have this code:

    if token.str != 'for' and token.str != 'while':
        continue

You can safely remove these 2 lines of code.